completions/dpkg

```diff
@@ -1,37 +1,65 @@
 # dpkg(1) and related commands completion                  -*- shell-script -*-
 
-if _comp_have_command grep-status; then
-    _comp_xfunc_dpkg_installed_packages()
-    {
-        grep-status -P -e "^${1-}" -a -FStatus 'ok installed' -n -s Package
-    }
-
-    _comp_xfunc_dpkg_purgeable_packages()
-    {
-        grep-status -P -e "^${1-}" -a -FStatus 'ok installed' -o -FStatus 'ok config-files' -n -s Package
-    }
-else
-    _comp_xfunc_dpkg_installed_packages()
-    {
+# @since 2.12
+_comp_xfunc_dpkg_installed_packages()
+{
+    local pkgs
+    pkgs=$(
+        grep-status -P -e "^${cur-}" -a \
+            -FStatus 'ok installed' \
+            -n -s Package 2>/dev/null
+    ) || pkgs=$(
         command awk -F '\n' -v RS="" "
-            index(\$1, \"Package: ${1-}\") == 1 &&
+            index(\$1, \"Package: ${cur-}\") == 1 &&
             \$2 ~ /ok installed|half-installed|unpacked|half-configured|^Essential: yes/ {
                 print(substr(\$1, 10));
             }" /var/lib/dpkg/status 2>/dev/null
-    }
+    )
+    _comp_compgen -a -- -W "$pkgs"
```

Comment:
```suggestion
    _comp_compgen -a -- -W '$pkgs'
```

As mentioned in https://github.com/scop/bash-completion/issues/958#issuecomment-1537214890, the `-W` option performs the expansions in the option argument, we would like to avoid the unexpected expansions that might be mixed in the outputs.